src/hocs/with-mouse-down-state/with-mouse-down-state.js

```diff
@@ -1,19 +1,36 @@
-import {
-  compose,
-  withState,
-  withHandlers,
-  wrapDisplayName,
-  setDisplayName,
-} from 'recompose';
+import React from 'react';
+import wrapDisplayName from '../wrap-display-name';
 
 export const stateHandlers = {
   handleMouseDown: ({ setMouseDown }) => () => setMouseDown(true),
   handleMouseUp: ({ setMouseDown }) => () => setMouseDown(false),
 };
 
-export default component =>
-  compose(
-    setDisplayName(wrapDisplayName(component, 'WithMouseDownState')),
-    withState('isMouseDown', 'setMouseDown', false),
-    withHandlers(stateHandlers)
-  )(component);
+const withMouseDownState = BaseComponent => {
+  class WithMouseDownState extends React.Component {
+    static displayName = wrapDisplayName(BaseComponent, 'withMouseDownState');
+
+    state = {
+      isMouseDown: false,
+    };
+
+    setMouseDown = nextMouseDown =>
+      this.setState({ isMouseDown: nextMouseDown });
+
+    handleMouseDown = () => this.setMouseDown(true);
+    handleMouseUp = () => this.setMouseDown(false);
+
+    render() {
+      return React.createElement(BaseComponent, {
```

Comment:
Any reason why using `createElement` over `<BaseComponent {...props} />`?

Reply:
I like `createElement`, I have no idea why. I can change it.

Reply:
I don't mind, I was just curious if there was a specific reason 😇